src/schedule/ll_schedule.c

```diff
@@ -380,7 +380,7 @@ static int schedule_ll_domain_set(struct ll_schedule_data *sch,
 
 	task_start_us = period ? period : start;
 	task_start_ticks = domain->ticks_per_ms * task_start_us / 1000;
-	task_start = task_start_ticks + k_cycle_get_64_atomic();
+	task_start = task_start_ticks + sof_cycle_get_64_atomic();
```

Comment:
Here we have complicated rollover assumptions that would break if we changed to 32bit here.